e-commerce/src/features/bookDetail/SelectedBook.tsx

```diff
@@ -0,0 +1,81 @@
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+import './SelectedBook.css';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { updateTitle } from '../title/titleSlice';
+import { addToCart ,cartItemType} from '../cart/cartSlice';
+import { useEffect } from 'react';
+import { getBookDetailRequest } from '../bookDetail/bookDetailSlice';
+
+
+function SelectedBook(){
+
+    const {state}=useLocation();
+    const params=useParams();
+    const navigate=useNavigate();
+    const dispatch = useAppDispatch();
+    
+    let bookId = params.id;
+
+    const bookDetail=useAppSelector(state=>state.bookDetail);
+    console.log("Book Detail is",bookDetail);
+
+    useEffect(()=>{
+        dispatch(getBookDetailRequest(bookId));
+    },[dispatch]);
+
+    console.log("Book Detail is2",bookDetail.books);
+
+    const imgSrc=bookDetail.books[0]?.thumbnail || "";
+    const bookTitle=bookDetail.books[0]?.title || "";
+    const authors=bookDetail.books[0]?.author || "";
+    const bookPrice=bookDetail.books[0]?.price || 0;
+    const pageCount=bookDetail.books[0]?.pageCount || 0;
+    const bookDescription=bookDetail.books[0]?.description || "";
+
+    dispatch(updateTitle(bookTitle));
+
+    function addToBag() : void{
+        const obj:cartItemType={
+            title:bookTitle,
+            price:bookPrice,
+            quantity:1,
+            author:authors,
+            imgSource:imgSrc
+        }
+        dispatch(addToCart(obj));
+    }
+
+    function buyNow() : void{
+        const obj:cartItemType={
+            title:bookTitle,
+            price:bookPrice,
+            quantity:1,
+            author:authors,

```

Comment:
looks like redundant code, introduce a common function that returns the data that is getting passed in the same schema structure